installkernel

```diff
@@ -78,9 +78,16 @@ updatever() {
 # into kernel package installation.  Also make sure the PATH includes
 # /usr/sbin and /sbin, just as dpkg would.
 if [ "${dir}" = "/boot" ]; then
-	PATH="${PATH}:/usr/sbin:/sbin" \
-		run-parts --verbose --exit-on-error --arg="${ver}" --arg="${img}" \
-		/etc/kernel/preinst.d
+	(
+		export LC_ALL=C PATH="${PATH}:/usr/sbin:/sbin"
+
+
+		for x in /etc/kernel/preinst.d/*; do
+			[ -x "${x}" ] || continue
+			echo "running ${x} ${ver} ${img}"
+			"${x}" ${ver} "${img}"
```

Comment:
This should exit 1 if there is some failure in one of the plugins

Reply:
The script is using `set -e`.

Reply:
The script explicitly runs under set -e though. The previous code didn't try to exit 1, it only told run-parts to stop running additional plugins if one fails.

Reply:
Right, makes sense.